src/sparsemax.jl

```diff
@@ -0,0 +1,61 @@
+"""
+    sparsemax(x; dims = 1)
+
+[Sparsemax](https://arxiv.org/abs/1602.02068) turns input array `x`
+into sparse probability distributions that sum to 1 along the dimensions specified by `dims`.
+
+Similar to softmax, each dimension is considered independent. For a matrix input `x` it will 
+by default (`dims = 1`) treat it as a batch of vectors, with each column independent. Keyword 
+`dims = 2` will instead treat rows independently.
+
+# Examples
+
+```jldoctest
+julia> sparsemax([1 2 3; 2 2 2])  # dims=1
+2×3 Matrix{Float64}:
+0.0  0.5  1.0
+1.0  0.5  0.0
+
+julia> sparsemax([1 2 3; 2 2 2]; dims=2)
+2×3 Matrix{Float64}:
+0.0       0.0       1.0
+0.333333  0.333333  0.333333
```

Comment:
These examples could come from a much simpler function. An idea for how to show that scale matters but shift doesn't is: 
```suggestion
julia> sparsemax(reshape(1:15, 5,3) ./ 2)
5×3 Matrix{Float64}:
 0.0   0.0   0.0
 0.0   0.0   0.0
 0.0   0.0   0.0
 0.25  0.25  0.25
 0.75  0.75  0.75

julia> sparsemax(reshape(1:15, 5,3) ./ 20)
5×3 Matrix{Float64}:
 0.1   0.1   0.1
 0.15  0.15  0.15
 0.2   0.2   0.2
 0.25  0.25  0.25
 0.3   0.3   0.3
```